geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/LTrimDUnitTest.java

```diff
@@ -0,0 +1,178 @@
+    AtomicBoolean isRunning = new AtomicBoolean(true);
+    List<String> listHashtags = makeListHashtags();
+    List<String> keys = makeListKeys(listHashtags);
+
+    List<String> elementList1 = makeElementList(keys.get(0), INITIAL_LIST_SIZE);
+    List<String> elementList2 = makeElementList(keys.get(1), INITIAL_LIST_SIZE);
+    List<String> elementList3 = makeElementList(keys.get(2), INITIAL_LIST_SIZE);
+
+    Runnable task1 =
+        () -> ltrimPerformAndVerify(keys.get(0), isRunning, elementList1);
+    Runnable task2 =
+        () -> ltrimPerformAndVerify(keys.get(1), isRunning, elementList2);
+    Runnable task3 =
+        () -> ltrimPerformAndVerify(keys.get(2), isRunning, elementList3);
+
+    Future<Void> future1 = executor.runAsync(task1);
+    Future<Void> future2 = executor.runAsync(task2);
+    Future<Void> future3 = executor.runAsync(task3);
+
+    for (int i = 0; i < 100; i++) {
+      clusterStartUp.moveBucketForKey(listHashtags.get(i % listHashtags.size()));
+      Thread.sleep(200);
+    }
+
+    isRunning.set(false);
+
+    future1.get();
+    future2.get();
+    future3.get();
+  }
+
+  private List<String> makeListHashtags() {
+    List<String> listHashtags = new ArrayList<>();
+    listHashtags.add(clusterStartUp.getKeyOnServer("ltrim", 1));
+    listHashtags.add(clusterStartUp.getKeyOnServer("ltrim", 2));
+    listHashtags.add(clusterStartUp.getKeyOnServer("ltrim", 3));
+    return listHashtags;
+  }
+
+  private List<String> makeListKeys(List<String> listHashtags) {
+    List<String> keys = new ArrayList<>();
+    keys.add(makeListKeyWithHashtag(1, listHashtags.get(0)));
+    keys.add(makeListKeyWithHashtag(2, listHashtags.get(1)));
+    keys.add(makeListKeyWithHashtag(3, listHashtags.get(2)));
+    return keys;
+  }
+
+  private void lpushPerformAndVerify(String key, List<String> elementList) {
+    jedis.lpush(key, elementList.toArray(new String[] {}));
+
+    Long listLength = jedis.llen(key);
+    assertThat(listLength).as("Initial list lengths not equal for key %s'", key)
+        .isEqualTo(elementList.size());
+  }
+
+  private void ltrimPerformAndVerify(String key,
+      AtomicBoolean isRunning,
+      List<String> elementList) {
+    while (isRunning.get()) {
+      lpushPerformAndVerify(key, elementList);
+
+      for (int i = 1; i <= INITIAL_LIST_SIZE / 2 && isRunning.get(); i++) {
+        long lastIndex = jedis.llen(key) - 2;
+        try {
+          assertThat(jedis.lindex(key, 0)).isEqualTo(makeElementString(key, INITIAL_LIST_SIZE - i));
+          assertThat(jedis.lindex(key, lastIndex)).isEqualTo(makeElementString(key, i));
+          jedis.ltrim(key, 1, lastIndex);
+          assertThat(jedis.llen(key)).as("Key: %s ", key).isEqualTo(lastIndex);
+        } catch (Exception ex) {
+          isRunning.set(false); // test is over
+          throw new RuntimeException("Exception performing LTRIM for list '"
+              + key + "' at step " + i + ": " + ex.getMessage());
+        }
+      }
```

Comment:
Exceptions thrown by the `assert`s won't be caught by the `catch` block and so won't stop the test. Not critical, but you could change this to:
```
        try {
          assertThat(jedis.lindex(key, 0))
              .as("lpush head verification failed at iteration " + i)
              .isEqualTo(makeElementString(key, INITIAL_LIST_SIZE - i));
          assertThat(jedis.lindex(key, lastIndex))
              .as("lpush tail verification failed at iteration " + i)
              .isEqualTo(makeElementString(key, i));
          jedis.ltrim(key, 1, lastIndex);
          assertThat(jedis.llen(key)).as("Key: %s ", key).isEqualTo(lastIndex);
        } catch (Throwable ex) {
          isRunning.set(false); // test is over
          throw ex;
        }
```

Reply:
Now I get what you mean by the throwable comment earlier. Done.